File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/trending/TrendingPaths.java
package dev.reuss.tmdb.domain.trending;


import dev.reuss.tmdb.domain.trending.model.TrendingTimeWindow;

final class TrendingPaths {

    private static final String TRENDING = "/trending";

    private TrendingPaths() {
    }

    static String movies(TrendingTimeWindow timeWindow) {
        return TRENDING + "/movie/" + timeWindow.value();
    }

    static String tvShows(TrendingTimeWindow timeWindow) {
        return TRENDING + "/tv/" + timeWindow.value();
    }

    static String people(TrendingTimeWindow timeWindow) {
        return TRENDING + "/person/" + timeWindow.value();
    }

    static String all(TrendingTimeWindow timeWindow) {
        return TRENDING + "/all/" + timeWindow.value();
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/trending/TrendingPaths.kt
package dev.reuss.tmdb.domain.trending

import dev.reuss.tmdb.core.path.tmdbPath
import dev.reuss.tmdb.domain.trending.model.TrendingTimeWindow

internal object TrendingPaths {
    fun movies(timeWindow: TrendingTimeWindow): String = tmdbPath("trending", "movie", timeWindow.value)

    fun tvShows(timeWindow: TrendingTimeWindow): String = tmdbPath("trending", "tv", timeWindow.value)

    fun people(timeWindow: TrendingTimeWindow): String = tmdbPath("trending", "person", timeWindow.value)

    fun all(timeWindow: TrendingTimeWindow): String = tmdbPath("trending", "all", timeWindow.value)
}
